public/app/plugins/panel/geomap/types.ts

```diff
@@ -49,4 +49,13 @@ export interface GeomapPanelOptions {
   controls: ControlsOptions;
   basemap: MapLayerConfig;
   layers: MapLayerConfig[];
+  fieldMapping: FieldMappingOptions;
+}
+
+export interface FieldMappingOptions {
+  queryFormat: string;
+  metricField: string;
+  geohashField: string;
+  latitudeField: string;
+  longitudeField: string;
```

Comment:
can these be part of the *layer* config, not the plugin config?  Different layers will have different mapping needs -- we can make it easy to reuse the patterns, but lets keep it off the root object.

Also `queryFormat` feels different than the other fields :)